### DISCUSSION TITLE
waltid-web-wallet sign-up error

### DISCUSSION THREAD

**Discussion by leandro-navarro** [walt-id/.github#67] [Help]
Hi, I tried running the [login-with-iota](https://docs.walt.id/v/web-wallet/ecosystems/iota/tutorials/login-with-iota) tutorial. 
When installing waltid-web-wallet, in step 4 I had to do: "cd web; yarn dev" [I'm running it in Macos 14, and node v19.9.0]
The server started (Local: http://localhost:3000/ ) and the Login page appeared in the browser. When trying to sign up for a new account I got an error popup window and this message on the backend in every signup try.

 ERROR  [h3] [unhandled] connect ECONNREFUSED ::1:4545 
  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1532:16)

Any suggestion? Thanks!

**Comment by taminobaumann** [comment]
Hey @leandro-navarro, 

thanks for reaching out! As the IOTA network changed with the latest protocol upgrade, our did registry with IOTA does not work at the moment. 

Please refer to our [roadmap](https://walt-id.notion.site/fcde1687baab42378b3047d4a22eeaca?v=1140dd17c17b4726a70cc1465d20866d) for upcoming IOTA protocol upgrades

**Reply by leandro-navarro** [reply]
The login with IOTA was the context, but the issue was with running and setting up a new account on waltid-web-wallet. Signing up for an account resulted in an error on the web page and an error message on the web-wallet container: "ERROR [h3] [unhandled] connect ECONNREFUSED ::1:4545". Do you recommend any specific node version to run waltid-web-wallet? Any clue about what service should run on port 4545? Thanks !

**Comment by leandro-navarro** [comment]
The error message on the web interface for signup is:
Error
{ "statusCode": 500, "stack": [] }

In the signup.vue file is related to:
async function submit() {
    isProgress.value = true

    const user = {
        username: emailInput,
        password: passwordInput,
        type: "email"
    }
    await $fetch("/r/auth/create", {
        method: 'POST',
        body: user
    })

When called on the terminal, I get the same error:
curl -X POST http://localhost:3000/r/auth/create -d '{
        username: "email@me",
        password: "abcde",      
        type: "email"
    }'    
{
  "statusCode": 500,
  "stack": []
}

which results in this log error in the backend:
ERROR  [h3] [unhandled] connect ECONNREFUSED ::1:4545                                                                                                          1:33:00 PM

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1532:16)

**Reply by leandro-navarro** [reply]
Sorry, I can see my installation was probably incomplete. 
I followed instead [the instructions in](https://docs.walt.id/v/apps/solutions/web-wallet/quick-start) to start with docker-compose instead of the yarn install path. This time, when visiting http://localhost:3000, I got a "Bad gateway" error, and these log entries that sound suspicious:
_docker-compose-wallet-backend-1   | 42 [main] INFO id.walt.Application - Starting walt.id wallet...
docker-compose-wallet-backend-1   | 51 [main] DEBUG id.walt.Application - Running in path: /waltid-web-wallet/.
docker-compose-wallet-backend-1   | 52 [main] INFO id.walt.Application - Setting up...
docker-compose-wallet-frontend-1  | Segmentation fault
docker-compose-wallet-frontend-1 exited with code 139
...
docker-compose-ingress-1          | 2023/10/12 12:11:26 [error] 7#7: *6 connect() failed (113: Host is unreachable) while connecting to upstream, client: 192.168.65.1, server: , request: "GET / HTTP/1.1", upstream: "http://172.18.0.6:3000/", host: "localhost:3000"_

I see port 4545 was not there before:
_docker-compose-wallet-backend-1   | 5709 [DefaultDispatcher-worker-4] INFO ktor.application - Responding at http://0.0.0.0:4545_
